content/blogs/gui-setup-part1-setupapi.md

```diff
@@ -0,0 +1,47 @@
+---
+title:      "1st-stage GUI setup - September 2023: partly Wine-syncing setupapi"
+author:     "hbelusca"
+date:       2023-10-30
+tags:       [ newsletter, gui-setup ]
+---
+
+Greetings to all ReactOS followers!
+As many of you certainly know by now, I have been officially hired by the ReactOS Deutschland e.V. to develop the graphical version of the 1st-stage ReactOS installer ("1st-stage GUI setup").
+
+During this first month (September 2023), my goal was to partly sync the code of the setupapi.dll module with Wine's.
+Indeed, this module was [forked](https://git.reactos.org/?p=reactos.git;a=blob;f=media/doc/WINESYNC.txt;h=f6614ac83a968cd5d7ddde6483dc85b94d264fc4;hb=HEAD#l182) way back in 2005, with later partial synchronizations of its code for some of its functions.
+
+For those of you who don't know what [setupapi](https://learn.microsoft.com/en-us/windows-hardware/drivers/install/setupapi) is, it is a module that: (i) exposes [generic setup functions](https://learn.microsoft.com/en-us/windows/win32/setupapi/setup-api-portal) for reading and processing INF files, moving/copying files from an installation source media to a target..., supporting also extraction from compressed .CAB cabinet files; and (ii) exposes device installation functions as well.
+
+Because of its very nature, it is "one of those Wine components we don't share completely because of architectural differences" [(CORE-7662)](https://jira.reactos.org/browse/CORE-7662) (due mainly to the device installation functions), and as such, as time goes by, Wine code improvements for code that isn't tied to OS- and architecture-specific aspects might go unnoticed and our shared code slowly drifts apart for no good reason.
+
+As part of the 1st-stage GUI setup, I need to use the [File Queues](https://learn.microsoft.com/en-us/windows/win32/setupapi/file-queues) and [Cabinet Files](https://learn.microsoft.com/en-us/windows/win32/setupapi/cabinet-files--setup-api-) functions. These are those that have received recent improvements/bug-fixes from Wine, and that can be synchronized without too much trouble.
+(NOTE: It should be observed that our current text-mode 1st-stage installer, USETUP, re-implements a very small subset of these functions, used for extracting files from the reactos.cab source file and copying them to the ReactOS installation directory. In the GUI setup, my aim is to use directly those exposed by setupapi.dll.)
```

Comment:
Starting a new line after each full-stop makes it easier to leave comments per-line